### PR TITLE
throw an exception when the route name is not found

### PR DESCRIPTION
**Pull request by AndreaCrotti**

### CODE REVIEW

File: core/src/martian/core.cljc [hunk 1]
```diff
@@ -46,7 +46,11 @@
     :else m))
 
 (defn find-handler [handlers route-name]
-  (first (filter #(= (keyword route-name) (:route-name %)) handlers)))
+  (let [handler (first (filter #(= (keyword route-name) (:route-name %)) handlers))]
```
Comment (oliyh) [inline core/src/martian/core.cljc:49]: Slightly redundant `let`?

File: core/test/martian/core_test.cljc [hunk 2]
```diff
@@ -311,9 +311,27 @@
            (martian/request-for m :create-blob {::martian/body body})
            (martian/request-for m :create-blob {:blob body})))))
 
+(deftest missing-route-test
+  (let [m (martian/bootstrap "https://camels.org"
+                             [{:route-name :create-camel
```
Comment (oliyh) [inline core/test/martian/core_test.cljc:316]: Could slim down this test by just giving an empty array for the routes

File: core/test/martian/core_test.cljc [hunk 3]
```diff
@@ -311,9 +311,27 @@
...
+                                 :query-schema {:camelVersion s/Int}
+                                 :body-schema {:Camel {:camelName s/Str
+                                                       :camelTrain {:leaderName s/Str
+                                                                    (s/optional-key :followerCamels) [{:followerName s/Str}]}
+                                                       :anyCamel s/Any}}
+                                 :headers-schema {(s/optional-key :camelToken) s/Str}
+                               :form-schema {:camelHumps (s/maybe s/Int)}}])]
+    (try
+      (martian/url-for m :missing-route {:camel-id 1})
+      (catch Exception e
```
Comment (oliyh) [inline core/test/martian/core_test.cljc:329]:
```suggestion
      (catch Throwable e
```

`Throwable` is defined at the top in a clj/cljs friendly way, this is currently failing on CI on js

**Review by oliyh — COMMENTED** [review]
Thanks for this. I agree this is a good way to go, although it is a breaking change to the API so will need to ensure this goes in the release notes as such.

File: core/src/martian/core.cljc (hunk 1 above)
Comment (AndreaCrotti) [reply core/src/martian/core.cljc]: Yes fixed

File: core/test/martian/core_test.cljc (hunk 2 above)
Comment (AndreaCrotti) [reply core/test/martian/core_test.cljc]: Done

**Pull request merged**